File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int cnt = 0;
        ListNode current = head;
        while(current != null){
            cnt++;
            current = current.next;
        }
        if(cnt == n){
            return head.next;
        }
        current = head;
        for (int i = 1; i < cnt - n; i++) {
            current = current.next;
        }
        if (current != null && current.next != null) {
            current.next = current.next.next;
        }

        return head;
    }
}

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head == NULL){
            return head;
        }
       
        ListNode* temp = head;
        ListNode* prev = NULL;
        int cnt = 0;
        while(temp != NULL){
            cnt++;
            prev = temp;
            temp = temp->next;
        }
        temp = head;
         if(n ==cnt){
            ListNode *temp = head;
            head = head->next;
            delete temp;
            return head;
        }
        for(int i = 1;i<cnt-n;i++){
            temp = temp->next;
        }
       if(temp != NULL && temp->next!=NULL){
            temp->next = temp->next->next;
            // delete temp;
        }
        return head;
    }
};